src/util/macros.lzz

```diff
@@ -64,7 +64,7 @@ void ThrowRangeError(const char* message) { EasyIsolate; isolate->ThrowException
 	if (db->busy)                                                              \
 		return ThrowTypeError("This database connection is busy executing a query")
 #define REQUIRE_DATABASE_NO_ITERATORS(db)                                      \
-	if (db->iterators)                                                         \
+	if (db->iterators && !db->unsafe_mode)                                                         \
```

Comment:
I think it makes sense that `unsafe` also affects the case above, not just the `iterators` case. E.g. to allow things such as #338 

And what about `locked`? Which use-case does that prevent and could this be disabled in unsafe mode as well?

Reply:
Please provide a test case which demonstrates the functionality and I'll include it.